File: BusReservations/BusReservations.Core/Domain/SeatModel/Seat.cs
namespace BusReservations.Core.Domain.SeatModel
{
    public class Seat
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public virtual Status Type { get; set; }
        public virtual float Discount { get; set; }
        public int SeatNumber { get; set; }

        public Seat()
        {

        }
        public Seat(int seatNumber, Status type)
        {
            SeatNumber = seatNumber;
            Type = type;
            switch (Type)
            {
                case Status.student:
                    Discount = 25;
                    break;
                case Status.elderly:
                    Discount = 50;
                    break;
                default:
                    Discount = 0;
                    break;
            }
        }
    }
}

File: Frontend/BusReservationsFrontend/src/app/models/seat.ts
import { Status } from './status';

export class Seat {
  seatNumber?: number;
  status: number;
  discount?: number;
  constructor(seatNumber: number, seatType: Status) {
    this.seatNumber = seatNumber;
    this.status = seatType;
    switch (seatType) {
      case Status.student:
        this.discount = 25;
        break;
      case Status.elderly:
        this.discount = 50;
        break;
      default:
        this.discount = 0;
        break;
    }
  }
}
